src/runtime-rs/crates/hypervisor/src/qemu/inner.rs

```diff
@@ -113,6 +118,15 @@ impl QemuInner {
         let mut command = Command::new(&self.config.path);
         command.args(cmdline.build().await?);
 
+        // The qemu process needs be moved into Network Namespace.
+        unsafe {
+            let _pre_exec = command.pre_exec(move || {
+                let _ = do_add_netns(&netns);
+
+                Ok(())
+            });
+        }
+
```

Comment:
Given the code as it is now, this is actually superfluous since the previous hunk already makes the executing thread enter the netns.

However, if we can manage to import `NetnsGuard` as proposed above, I would like to use it to enter the netns _just_ for the `add_network_device()` invocation.  In that case, this hunk would become useful.